src/persistence/inventory.rs

```diff
@@ -301,22 +344,39 @@ pub trait Inventory: Deref<Target = Self::Stash> {
     where
         R::Error: 'static;
 
-    /// # Safety
+    /// Imports fascia into the stash, index and inventory.
     ///
-    /// Assumes that the bundle belongs to a non-mined witness transaction. Must
-    /// be used only to consume locally-produced bundles before witness
-    /// transactions are mined.
-    fn consume_anchor(
+    /// Part of the transfer workflow. Called once PSBT is completed and an RGB
+    /// fascia containing anchor and all state transitions is exported from
+    /// it.
+    ///
+    /// Must be called before the consignment is created, when witness
+    /// transaction is not yet mined.
+    fn consume(&mut self, _fascia: Fascia) -> Result<(), InventoryError<Self::Error>> {
+        todo!();
+        /*
+        let witness_id = fascia.anchor.witness_id();
+        unsafe { self.consume_anchor(fascia.anchor)? };
+        for bundle in fascia.bundles {
+            let ids1 = bundle.known_transitions.keys().copied().collect::<BTreeSet<_>>();
+            let ids2 = bundle.input_map.values().copied().collect::<BTreeSet<_>>();
+            if !ids1.is_subset(&ids2) {
+
+            }
+            unsafe { self.consume_bundle(fascia.contract_id, bundle, witness_id)? };
+        }
+        Ok(())
+         */
+    }
+
+    #[doc(hidden)]
+    unsafe fn consume_anchor(
         &mut self,
-        anchor: Anchor<mpc::MerkleBlock>,
+        anchor: XAnchor<mpc::MerkleBlock>,
     ) -> Result<(), InventoryError<Self::Error>>;
 
-    /// # Safety
```

Comment:
Can you re-add the docs for the unsafe code? Just indicating why unsafe and under what conditions

Reply:
I removed it so it since it will not be seen anyway due to the added `doc(hidden)` method... I do not know how to hide such things properly and at the same time provide those implementing the trait with enough info...